internal/yarpcerrors/yarpcerrors.go

```diff
@@ -31,3 +35,16 @@ import "go.uber.org/yarpc/yarpcerrors"
 func NewWithNamef(code yarpcerrors.Code, name string, format string, args ...interface{}) *yarpcerrors.Status {
 	return yarpcerrors.Newf(code, format, args...).WithName(name)
 }
+
+// AnnotateWithInfo will take an error and add info to it's error message while
+// keeping the same status code.
+func AnnotateWithInfo(status *yarpcerrors.Status, format string, args ...interface{}) *yarpcerrors.Status {
+	return yarpcerrors.Newf(status.Code(), "%s: %s", sprintf(format, args...), status.Message())
+}
+
+func sprintf(format string, args ...interface{}) string {
```

Comment:
Is this necessary? Does fmt.Sprintf just do this?

Reply:
huh, yeah, just copied it from the errors api, removing.